Push changes

tomczoink opened the pull request:
Updates to Paddy's current PR, in addition to new pages in the realtime and rest sections.

mattheworiordan commented on content/general/push/activate-subscribe.textile:
```diff
@@ -373,3 +373,57 @@ blang[android].
+    default: PushChannel
+    ruby:    Ably::Models::PushChannel
+
+<%= partial 'types/_push_channel' %>
```
I assume you'll need to update these all to use `partial_version` once all the work is merged.

mattheworiordan commented on content/general/push/publish.textile:
```diff
@@ -141,12 +141,16 @@ channel.publish({ name: 'example', data: 'data', extras: extras }, function(err)
 
 See the "push admin publish documentation":/general/push/admin#publish for the client library API details, and the "raw push publish REST API documentation":/rest-api#push-publish for information on the underlying direct publishing endpoint used by the client libraries.
 
+Notably, the "push admin":/general/push/admin is only available in our Ruby and JavaScript libraries. If you need to publish to devices with another library, you should use "channel-based publishing":#channel-broadcast.
```
Sadly, this is not true.  It is supported in pretty much all of our libraries now, PHP, Python, Node, JS, Ruby, iOS, Android, Java etc.  The only libs that don't support it are Go and .NET, and the latter will add support soonish.

mattheworiordan commented on content/realtime/push.textile:
```diff
@@ -0,0 +1,56 @@
+
+- "Apple Push Notifications":https://developer.apple.com/notifications/ := supported on all mobile devices running iOS and desktop devices running OS X
+- "Firebase Cloud Messaging":https://firebase.google.com/docs/cloud-messaging/ := supported on all Android and iOS devices, although we use FCM exclusively for Android message delivery
+- Experimental "W3C Push API":https://www.w3.org/TR/push-api/ := experimental support for "modern W3C compliant browsers":https://caniuse.com/#feat=push-api (this does not include Apple's Safari browser)
```
Perhaps add a "get in touch" if you want to use the experiemental browser support.

mattheworiordan commented on content/rest/push.textile:
```diff
@@ -0,0 +1,56 @@
+---
```
THis content is duplicated, should we use a partial to avoid these diverging in future?

tomczoink commented:
Updated per feedback @mattheworiordan, added other languages besides C# and Go to admin. @paddybyers if you could also give a review it'd be much appreciated.

paddybyers commented on content/general/push/_push_intro.textile:
```diff
@@ -0,0 +1,45 @@
+
+Every device that will receive push notifications must activate itself with the local operating system or framework, and hook into the push notification services that the underlying platform provides. This functionality is platform-specific and can also vary considerably across not just platforms, but also across the push services that operate on those platforms such as GCM and FCM, both of which ...
+
+The Ably client libraries aim to abstract away this complexity and platform-specific behaviour by providing a consistent API for device activation, maintenance of the device registration, subscription to Ably channels for receiving push notifications, and of course for receiving the native push notifications themselves.
```
We don't abstract "receiving the native push notifications themselves". Apps will still use the existing native platform APIs for this.

paddybyers commented on content/general/push/_push_intro.textile:
```diff
@@ -0,0 +1,45 @@
+
+Whilst the realtime client libraries provide APIs for a device to activate itself (via "@client.push@":/general/push/activate-subscribe) and subscribe for push notifications (via "@channel.push@":/general/push/activate-subscribe), those APIs are intentionally limited to actions pertaining to the device it is run on.
+
+A separate and distinct push admin API is additionally provided in our client libraries specifically designed for use by your servers to facilitate managing and delivering push notifications across all of your registered devices. This API, amongst other things, includes features to manage registered devices, channel subscriptions and deliver push notifications directly. Currently the "push admin  ...
```
Push admin is supported by JS, Ruby, Java/Android and iOS

paddybyers commented on content/general/push/activate-subscribe.textile:
```diff
@@ -15,9 +15,11 @@ jump_to:
+Every device that will receive push notifications must activate itself with the local operating system or framework, and hook into the push notification services that the underlying platform provides. This functionality is platform-specific and can also vary considerably across not just platforms, but also across the push services that operate on those platforms such as GCM and FCM, both of which ...
 
-The client libraries also provide a set of admin functionality that is typically used server-side (in a trusted environment) to manage all devices and push notification delivery and subscriptions. You can find out more in the "push admin documentation":admin.
+The Ably client libraries aim to abstract away this complexity and platform-specific behaviour by providing a consistent API for device activation, maintenance of the device registration, subscription to Ably channels for receiving push notifications, and of course for receiving the native push notifications themselves.
```
"and of course for receiving the native push notifications themselves": we don't do this bit
